tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -4269,7 +4269,7 @@ def __init__(self,
           f=self._map_func.function,
           deterministic=deterministic_string,
           **self._flat_structure)
-    elif deterministic is not None or compat.forward_compatible(2020, 2, 20):
+    else:
```

Comment:
You should probably retain `elif deterministic is not None:`

Reply:
This won't make any difference since `compat.forward_compatible(2020, 2, 20)` will always be `True` and `if deterministic is not None or True` therfore will evaluate to `True` regardless of the value of `deterministic`.